language/move-model/src/builder/module_builder.rs

```diff
@@ -1437,11 +1437,21 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
                 let first = exps.remove(0);
                 (first, exps)
             }
+            ConditionKind::Emits => {
+                let (_, first) = et.translate_exp_free(exp);
+                let vector_u8_type = Type::Vector(Box::new(U8_TYPE));
+                let mut exps =
+                    vec![et.translate_exp(additional_exps.get(0).unwrap(), &vector_u8_type)];
+                if additional_exps.len() > 1 {
+                    exps.push(et.translate_exp(additional_exps.get(1).unwrap(), &expected_type));
```

Comment:
Is the expected_type here the right one? It seems the function self.expected_type_for_condition(kind) is unchanged, this can be hardly right.

Reply:
I changed it to be BOOL_TYPE because it's the expression for "if".